version.rs

```diff
@@ -46,32 +57,61 @@ impl TryFrom<Deployment> for Entry {
     }
 }
 
-#[get("/versions")]
-async fn get_versions(store: Data<Store<Deployment>>) -> impl Responder {
+#[derive(Debug)]
+enum Error {
+    NotFound,
+}
+
+impl IntoResponse for Error {
+    type Body = Full<Bytes>;
+    type BodyError = Infallible;
+
+    fn into_response(self) -> Response<Self::Body> {
+        let (status, error_message) = match self {
+            Error::NotFound => (StatusCode::NOT_FOUND, "not found"),
+        };
+
+        let body = Json(json!({
+            "error": error_message,
+        }));
+
+        (status, body).into_response()
+    }
+}
```

Comment:
Is there a way to just return the 404 directly from the /versions handler? Trying to keep the file small since part of the selling point in the readme is that this thing is basically ~100 lines, so I don't think we need to concern ourselves too deeply with error handling here.

Reply:
I think so, but I will need to figure out an appropriate type signature because `impl IntoResponse` doesn't quite seem to cut it.

Reply:
[`(StatusCode, impl IntoResponse)`](https://docs.rs/axum/0.2.8/axum/response/trait.IntoResponse.html#impl-IntoResponse-for-(StatusCode%2C%20T)) should do it. Can also nest the types in many ways. For example `Result<Json<T>, (StatusCode, String)>`.

Reply:
5eb289fccbfdd910a40d1a2d8105a5e555af490d